aiutils/tftools/placeholder_management.py

```diff
@@ -0,0 +1,173 @@
+        """Add placeholder.
+        
+        If the sparse is True then 3 placeholders are automatically created 
+        corresponding to the indices and values of the non-zero entries, and 
+        shape of the sparse matrix. The user does not need to keep track of 
+        these and can directly pass a sparse scipy matrix as input and a 
+        Tensorflow SparseTensor object is made available for use in the graph.
+
+        Args:
+            name (str): Name of the placeholder. 
+            dtype (tf.Dtype): Data type for the placeholder.
+            shape (list of ints): Shape of the placeholder.
+            sparse (bool): Specifies if the placeholder takes sparse inputs.
+        """
+
+        self.issparse[name] = sparse
+        if not sparse:
+            self._placeholders[name] = tf.placeholder(dtype, shape, name)
+
+        else:
+            name_indices = name + '_indices'
+            name_values = name + '_values'
+            name_shape = name + '_shape'
+
+            self._placeholders[name_indices] = tf.placeholder(tf.int64, [None, 2],
+                                                           name_indices)
+            self._placeholders[name_values] = tf.placeholder(dtype, [None],
+                                                          name_values)
+            self._placeholders[name_shape] = tf.placeholder(tf.int64, [2],
+                                                         name_shape)
+
+    def __getitem__(self, name):
+        """Returns placeholder with the given name.
+        
+        Usage:
+            plh_mgr = PlaceholderManager()
+            plh_mgr.add_placeholder('var_name', tf.int64, sparse=True)
+            placeholder = plh_mgr['var_name']
+        """
+        sparse = self.issparse[name]
+        if not sparse:
+            placeholder = self._placeholders[name]
+        else:
+            placeholder_indices = self._placeholders[name + '_indices']
+            placeholder_values = self._placeholders[name + '_values']
+            placeholder_shape = self._placeholders[name + '_shape']
+            sparse_tensor = tf.SparseTensor(
+                placeholder_indices, placeholder_values, placeholder_shape)
+            placeholder = sparse_tensor
+
+        return placeholder
+
+    def get_placeholders(self):
+        """Returns a dictionary of placeholders with names as keys.
+        
+        The returned dictionary provides an easy way of refering to the 
+        placeholders and passing them to graph construction or evaluation 
+        functions.
+        """
+        placeholders = dict()
+        for name in self.issparse.keys():
+            placeholders[name] = self[name]
+
+        return placeholders
+
+    def get_feed_dict(self, inputs):
+        """Returns a feed dictionary that can be passed to eval() or run().
+
+        This method creates a feed dictionary from provided inputs that can be
+        passed directly into eval() or run() routines in Tensorflow. 
```

Comment:
Can you make an example usage e.g. "operation.eval(placeholder_manager.get_feed_dict({"x",x}))"  (this is the expected usage yes?)